### PR TITLE
CVE-2014-7935 and CVE-2013-2858

### PR DESCRIPTION
**Pull request by ConR24**
Added research for CVE-2014-7935 and CVE-2013-2858

### CODE REVIEW

File: cves/CVE-2013-2858.yml [hunk 1]
```diff
@@ -156,38 +175,41 @@ lessons:
...
   least_privilege:
-    applies: 
-    note: 
+    applies: false
+    note:
   frameworks_are_optional:
-    applies: 
-    note: 
+    applies: true
+    note: This vulnerability was caused by their usage of the WebKit library. If they didn't use this library
```
Comment (andymeneely) [inline cves/CVE-2013-2858.yml]: This is why the build is failing. Follow the YML syntax for multiline strings here (i.e. a pipe char). Don't forget that getting the CI to build for Part 1 is a big part of your grade.

File: cves/CVE-2014-7935.yml [hunk 2]
```diff
@@ -32,7 +32,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  A vulnerability with how memory associated with speech commands in
```
Comment (gldias) [inline cves/CVE-2014-7935.yml]: This sounds like a run-on sentence. It's not a big deal, but it could sound a bit better with more punctuation.

File: cves/CVE-2013-2858.yml [hunk 3]
```diff
@@ -32,13 +32,16 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  There is a use-after-free vulnerability with how html 5 audio is
```
Comment (gldias) [inline cves/CVE-2013-2858.yml]: This sounds like a really interesting vulnerability, but I feel that this description is a bit jargon-y. Try to reword it in a way that someone with little to no security knowledge would understand.

File: cves/CVE-2014-7935.yml [hunk 4]
```diff
@@ -76,9 +81,12 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    While I did find a unit testing file, it doesn't seem that it was involved with
```
Comment (angbamfield) [inline cves/CVE-2014-7935.yml]: Do you think they just tried to recreate the vulnerability to test out the fix? This is interesting why they didn't write any unit tests to check all the different edge cases.

File: cves/CVE-2014-7935.yml [hunk 5]
```diff
@@ -94,20 +102,25 @@ discovered:
...
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    A user discovered this vulnerability by opening an incognito session
+    of google chrome and went to a webpage that they created that allowed the
+    user to click a button that would simulate a speech command. Once the user
```
Comment (angbamfield) [inline cves/CVE-2014-7935.yml]: Wait they were able to click on any button that was on the webpage ?

File: cves/CVE-2014-7935.yml [hunk 6]
```diff
@@ -131,12 +145,16 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    I found a lot of just general updates to the chromium, but two major events I
+    found were them removing an entire library from their codebase and adding unused code for a
```
Comment (angbamfield) [inline cves/CVE-2014-7935.yml]: Interesting. Do you think they were trying to fix a different vulnerability in the process by adding this unused code?

File: cves/CVE-2014-7935.yml [hunk 7]
```diff
@@ -197,4 +219,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    After looking at the fix in the code, this is obviously a coding mistake, as they
```
Comment (angbamfield) [inline cves/CVE-2014-7935.yml]: Good review! Just mentioned earlier just make sure to fix any spelling/punctuation mistakes. Overall very interesting vulnerabilities !!

File: cves/CVE-2013-2858.yml [hunk 8]
```diff
@@ -4,21 +4,21 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: CWE-416 use-after-free
```
Comment (scioffi) [inline cves/CVE-2013-2858.yml]: The CWE value should just be the CWE ID number.
```suggestion
CWE: 416
```

File: cves/CVE-2014-7935.yml [hunk 9]
```diff
@@ -4,21 +4,21 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: CWE-416 use-after-free
```
Comment (scioffi) [inline cves/CVE-2014-7935.yml]: The CWE value should just be the CWE ID number.
```suggestion
CWE: 416
```

File: cves/CVE-2013-2858.yml (hunk 8 above)
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: Fixed

File: cves/CVE-2014-7935.yml (hunk 9 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: Fixed

File: cves/CVE-2013-2858.yml (hunk 3 above)
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: I fleshed this out some more to give more information and use less security terms, let me know if this is better

File: cves/CVE-2014-7935.yml (hunk 2 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: I re-wrote the description to better explain this vulnerability, let me know if that improves it.

File: cves/CVE-2013-2858.yml [hunk 10]
```diff
@@ -49,24 +52,24 @@ reviews:
...
+- :commit: 5189b4c748c173f48f0f82323ec27d7c2ef42d4a
+  :note: Added 2011-08-18 by crogers
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes:
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: Updated, thanks.

File: cves/CVE-2014-7935.yml [hunk 11]
```diff
@@ -46,24 +50,25 @@ reviews:
...
+  added on 2013-12-19 by dmazzoni'
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes:
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: Updated, thanks.

File: cves/CVE-2014-7935.yml (hunk 5 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: This was a webpage that the person who discovered the vulnerability made that only had one button for you to press, I'll update this to be more specific.

File: cves/CVE-2014-7935.yml [hunk 12]
```diff
@@ -46,24 +50,25 @@ reviews:
...
+  added on 2013-12-19 by dmazzoni'
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 2
```
Comment (gldias) [inline cves/CVE-2014-7935.yml]: +3 upvotes

File: cves/CVE-2014-7935.yml [hunk 13]
```diff
@@ -197,4 +219,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
```
Comment (gldias) [inline cves/CVE-2014-7935.yml]: I also see a few spelling/grammar mistakes in your files, so make sure you review them again and fix any mistakes you may find. Besides that, I thought these CVE's were really interesting!

File: cves/CVE-2013-2858.yml (hunk 3 above)
Comment (gldias) [reply cves/CVE-2013-2858.yml]: Looks good!

File: cves/CVE-2013-2858.yml [hunk 14]
```diff
@@ -49,24 +52,24 @@ reviews:
...
 - 239897
-repo: 
+repo:
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 74480e1627623f29217a4f3ee4a5ca2cc83951a8
-  :note: ''
+  :note: Fixed on 2018-05-14 by crogers
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: Sure thing I can add that

File: cves/CVE-2013-2858.yml [hunk 15]
```diff
@@ -49,24 +52,24 @@ reviews:
...
 vccs:
-- :commit:
-  :note: 
+- :commit: 5189b4c748c173f48f0f82323ec27d7c2ef42d4a
+  :note: Added 2011-08-18 by crogers
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: This is the commit where the vulnerability was introduced, therefore I don't believe anything related to how it was discovered should be here. I can add a description of what was added if that would make it better.

File: cves/CVE-2014-7935.yml (hunk 7 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: Thanks! I'll fix those spelling issues.

File: cves/CVE-2014-7935.yml (hunk 13 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: Thanks! I'll fix those spelling and grammar issues.

File: cves/CVE-2014-7935.yml (hunk 12 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: Updated, thanks.

File: cves/CVE-2013-2858.yml [hunk 16]
```diff
@@ -49,24 +53,24 @@ reviews:
...
 upvotes_instructions: |
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: Updated, thanks

File: cves/CVE-2014-7935.yml (hunk 6 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: I don't think they were, but rather they were starting work on a new feature when they added this unused code. Smart locking sounds like it was an extra layer of security that would check if your location was in a close proximity of a verified device when you tried to log in.

File: cves/CVE-2013-2858.yml [hunk 17]
```diff
@@ -120,12 +131,18 @@ interesting_commits:
     emerging themes?
 
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    The reason I find this commit so interesting is beacuse of how out of no
+    where it is when looking through the git log. In between the multiple commits
+    updating the webkit subsystem this entire library was just removed from the project.
+    This was a good idea for them to do because if they left this unused library in the code base, it could have lead to more
+    vulnerabilities.
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: After looking into it, it seems that the V8Proxy led to a lot of DoS vulnerabilities when it was in use because it would use up a lot of memory at times which is why it was unused. I still think that it's good to remove unused libraries because they might allow vulnerabilities that would allow things such as replicate the vulnerabilities that would occur when V8Proxy was being used or even other things such as remote code execution.

File: cves/CVE-2014-7935.yml (hunk 4 above)
Comment (ConR24) [reply cves/CVE-2014-7935.yml]: I think they did just try to recreate the vulnerability to test if it was fixed or not. I did some more digging into the subsystem and could not find any reference to the problem file being tested. I also found on the discussion for the bug report there are mentions of people being able to still reproduce the vulnerability, which backs up that they were testing it manually.

File: cves/CVE-2013-2858.yml [hunk 18]
```diff
@@ -79,9 +83,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    It seems that this file did not have any units tests involved with this vulenrability.
+    In the original vcc I found assert statements, but those were absent in the fix commit.
```
[the remark this answers is not in the document]
Comment (ConR24) [reply cves/CVE-2013-2858.yml]: It seems that they had been removed from the file somewhere in between the vcc and the fix commits. I think that these were asserts being used to test for a different vulnerability because they would assert a value that was the opposite of the check below it. They were probably removed once they were no longer needed, which is why they were not present in the code base of the fix commit.
